libvast/vast/detail/posix.hpp

```diff
@@ -18,6 +22,27 @@ namespace vast::detail {
 
 enum class socket_type { datagram, stream, fd };
 
+/// Holds the necessary state to send unix datagrams to a destination socket.
+struct uds_datagram_sender {
+  /// The destructor unlinks the src socket.
+  ~uds_datagram_sender();
+
+  /// Helper function to create a sender.
+  static caf::expected<uds_datagram_sender> make(const std::string& path);
```

Comment:
Is this not a `std::filesystem::path`?

Reply:
Yes it is, but the rest of `posix.hpp` doesn't use filesystem either, and it is better to keep local consistency.